geps/gep-2643/index.md

```diff
@@ -0,0 +1,560 @@
+  * A `Gateway listener` with `*.example.com` as the hostname matches a `TLSRoute` that
+  specifies at least one hostname that matches the `Gateway listener` hostname.
+  For example, `test.example.com` and `*.example.com` would both match. On the
+  other hand, `example.com` and `test.example.net` would not match.
+  * If both the `Gateway listener` and `TLSRoute` specify hostnames, any `TLSRoute`
+  hostnames that do not match the `Gateway listener` hostname MUST be ignored
+  for that Listener. For example, if a `Gateway listener` specified `*.example.com`,
+  and the `TLSRoute` specified `test.example.com` and `test.example.net`,
+  the later must not be considered for a match. In case none of the hostnames specified
+  on the `TLSRoute` matches the Gateway hostnames, the route MUST have a `Condition` of 
+  `Accepted=False` with the reason `NoMatchingListenerHostname`.
+  * In any of the cases above, the `TLSRoute` should have a `Condition` of `Accepted=True`.
+
+## Conformance Details
+
+###  Feature Names
+
+* SupportTLSRoute
+* SupportTLSRouteTermination
+* SupportTLSRouteMixedMode
+
+### Conformance tests
+
+| Description | Outcome | Features |
+| :---- | :---- | :---- |
+| A single TLSRoute in the gateway-conformance-infra namespace attaches to a Gateway in the same namespace. <br/> Code: [Simple] <br/>Issue: [TLSRoute conformance] | A request to a hostname served by the TLSRoute should be passthrough directly to the backend. Check if the termination happened, if no additional Gateway header was added. | SupportTLSRoute |
+| A single TLSRoute in the gateway-conformance-infra namespace, with a backendRef in another namespace without valid ReferenceGrant, should have the ResolvedRefs condition set to False. <br/> Code: [ReferenceGrant] and the request to the route MUST fail [Invalid ReferenceGrant Request] | TLSRoute conditions must have a `RefNotPermitted` status. A request to the route MUST fail | SupportTLSRoute + SupportReferenceGrant |
+| A TLSRoute trying to attach to a gateway without a “tls” listener MUST be rejected  | TLSRoute should have a parent condition of type `Accepted=False` with Reason `NoMatchingParent`. Request to the route MUST fail. | SupportTLSRoute |
+| A TLSRoute with a hostname that does not match the Gateway hostname MUST be rejected (eg.: route with hostname [www.example.com](http://www.example.com), gateway with hostname www1.example.com) <br/> Issue: [TLSRoute conformance] | Condition on the TLSRoute parent of type `Accepted=False` with Reason `NoMatchingListenerHostname`. Request to the route MUST fail. | SupportTLSRoute |
+| A Gateway containing a Listener of type TLS/Terminate SHOULD be accepted, and SHOULD direct the requests to the right TLSRoute when TLSRoute termination is supported. <br/> Issue: [Termination] | Being able to do a request to a TLS route being terminated on gateway (eg.: terminated.example.tld/xpto) | SupportTLSRoute + SupportTLSRouteTermination |
```

Comment:
Is this really "MUST be accepted if Termination is supported"? Should we phrase it that way?

(Same applies to the other cases discussing termination.)